src/ui/molecules/select/Select.tsx

```diff
@@ -0,0 +1,317 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import type { RefObject } from 'react'
+import classNames from 'classnames'
+import {
+  capitalizeFirstLetter,
+  capitalizeFirstLetterOfEachArrayWord,
+  compareStrings,
+  isString
+} from 'utils'
+import type { UseState } from 'superTypes'
+import { InputBase } from 'ui/atoms/inputBase/InputBase'
+import type { InputBaseProps } from 'ui/atoms/inputBase/InputBase'
+import { Typography } from 'ui/atoms/typography/Typography'
+import { Icon } from 'ui/atoms/icon/Icon'
+import short from 'short-uuid'
+import './select.scss'
+
+export type Option = {
+  readonly label: string
+  readonly value: string
+  readonly group?: string
+}
+
+type InputPropsForSelect = Pick<InputBaseProps, 'placeholder' | 'disabled' | 'hasError'>
+
+export type SelectProps = InputPropsForSelect & {
+  /**
+   * Defines the callback called when the select value changes.
+   */
+  readonly onChange: (value: string | Readonly<string[]>) => void
+  /**
+   * The value of the select.
+   */
+  readonly value: string | string[]
+  /**
+   * The options list displayed when the select is opened.
+   */
+  readonly options: Readonly<Option[]>
+  /**
+   * If true, allows the user to make a multiple choice.
+   * Default to false.
+   */
+  readonly multiple?: boolean
+  /**
+   * The size of the input field.
+   * It will be automatically adjusted responsively to the screen size.
+   */
+  readonly size?: 'x-large' | 'large' | 'medium' | 'small' | 'x-small'
+  /**
+   * If true, the select will take 100% of its parent's size.
+   */
+  readonly fullWidth?: boolean
+  /**
+   * Displays a message to the user below the select area.
+   */
+  readonly helperText?: string
+}
+
+// eslint-disable-next-line max-lines-per-function, @typescript-eslint/prefer-readonly-parameter-types
+export const Select = ({
+  placeholder,
+  size = 'small',
+  disabled = false,
+  hasError = false,
+  multiple = false,
+  options,
+  onChange,
+  fullWidth,
+  value,
+  helperText
+}: SelectProps): JSX.Element => {
```

Comment:
It's up to you but you could wrap the SelectProps in a deepReadonly and get rid of the Readonly in SelectProps. Makes the type a bit easier to read 😄